File: medium/diamond.js
const ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

export const rows = (letter) => {
  let diamond = [];
  let index = ALPHABET.indexOf(letter);
  for (let i = -index; i <= index; i++) {
    let row = "";
    for (let j = -index; j <= index; j++) {
      if (Math.abs(j) === -Math.abs(i) + index) {
        row += ALPHABET[-Math.abs(i) + index];
      } else {
        row += " ";
      }
    }
    diamond.push(row);
  }
  return diamond;
};

File: python/easy/diamond.py
ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

def rows(letter):
    diamond = []
    index = ALPHABET.index(letter)
    half_width = index + 1
    for i in range(half_width):
        line = ""
        for j in range(half_width):
            if (j + i) == index:
                line += ALPHABET[i]
            else:
                line += " "
        for j in range(index - 1, -1, -1):
            line += line[j]
        diamond.append(line)
    for i in range(index - 1, -1, -1):
        diamond.append(diamond[i])
    return diamond